### PR TITLE
xmlserializer: load Xsd Schema from xml attribute

### PR DESCRIPTION
**Pull request by miguelgaio**
xsi:noNamespaceSchemaLocation provides xsd location.
Use libxml2 API and the attribute to found out xsd schemas

Signed-off-by: Miguel Gaio <miguel.gaio@intel.com>
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/248%23issuecomment-142637504%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23issuecomment-143719469%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23issuecomment-143731574%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23issuecomment-143837318%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23issuecomment-144006164%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40217333%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40218494%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40229458%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40314448%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40314653%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40315214%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40342900%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40542611%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40542614%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40542618%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40542620%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40546172%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40547789%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/248%23issuecomment-142637504%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40dawagner%20%40krocard%20%40OznOg%20%40cc6565%20please%20review%22%2C%20%22created_at%22%3A%20%222015-09-23T15%3A20%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-28T11%3A39%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%2C%20%7B%22body%22%3A%20%22nice%20job%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-09-28T12%3A39%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40krocard%20or%20%40OznOg%20please%20review.%20I%27ll%20merge%20when%20there%20is%20another%20%2B1.%22%2C%20%22created_at%22%3A%20%222015-09-28T18%3A37%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-29T09%3A34%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9692938%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cc6565%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2049ffd604786e225e8396aa4a2aa13d658fd40f81%20parameter/ParameterMgr.cpp%2093%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40546172%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%5C%22URI%5C%22%20or%20%5C%22Uri%5C%22%20%3F%20Please%20choose%20one%20and%20stick%20to%20it.%20%28I%27d%20choose%20%5C%22Uri%5C%22%20because%20even%20though%20it%27s%20an%20acronym%2C%20we%20have%20chose%20lowerCamelCase%20style.%20EDIT%3A%20and%20because%20we%20already%20have%20%5C%22mk%2A%2AXml%2A%2ADoc%5C%22%29%22%2C%20%22created_at%22%3A%20%222015-09-28T12%3A34%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-09-28T12%3A54%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterMgr.cpp%3AL597-604%22%7D%2C%20%22Pull%20d14954fb9e808b40ee07233c4f805e8f3962e1b1%20xmlserializer/XmlDocSource.cpp%2078%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40217333%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22all%20those%20free%20could%20be%20removed%20with%20unique_ptr...%22%2C%20%22created_at%22%3A%20%222015-09-23T15%3A26%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%27m%20not%20sure%2C%20xmlFree%20is%20not%20delete%20or%20free%28%29%22%2C%20%22created_at%22%3A%20%222015-09-23T15%3A34%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%2C%20%7B%22body%22%3A%20%22unique_ptr%20can%20be%20provided%20with%20a%20custom%20deleter%20%28default%20is%20delete%2C%20not%20%5Cr%5Cnfree%29.%22%2C%20%22created_at%22%3A%20%222015-09-23T17%3A07%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-28T11%3A39%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20xmlserializer/XmlDocSource.cpp%3AL225-232%22%7D%2C%20%22Pull%207a207c7fd3341ebaf18501bf9f6b2f3d0efb0372%20test/CMakeLists.txt%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40314448%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22It%20could%20be%20worth%20copying%20this%20directory%20in%20the%20legacy%20functional%20test%20dir%20too%20%28or%20instead%29.%20And%20if%20so%2C%20it%20should%20be%20done%20in%20%60functional-tests-legacy/CMakeLists.txt%60.%5Cr%5Cn%5Cr%5CnYou%20would%20avoid%20the%20%60../../../../../../%60%20in%20the%20XML%20files%20below.%5Cr%5Cn%5Cr%5CnEDIT%3A%20and/or%20use%20%60configure_file%60%20on%20all%20the%20XML%20files%20contained%20in%20the%20legacy%20functional%20tests.%22%2C%20%22created_at%22%3A%20%222015-09-24T12%3A48%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-28T11%3A39%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/CMakeLists.txt%3AL26-34%22%7D%2C%20%22Pull%207a207c7fd3341ebaf18501bf9f6b2f3d0efb0372%20xmlserializer/XmlDocSource.h%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40315214%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20constructor%20seems%20unused.%20If%20so%2C%20you%20may%20remove%20it%20altogether%20%28in%20a%20separate%20patch%29.%22%2C%20%22created_at%22%3A%20%222015-09-24T12%3A57%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22Used%20by%20XmlMemorySource%22%2C%20%22created_at%22%3A%20%222015-09-24T16%3A53%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-28T11%3A39%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20xmlserializer/XmlDocSource.h%3AL57-75%22%7D%2C%20%22Pull%207a207c7fd3341ebaf18501bf9f6b2f3d0efb0372%20test/functional-tests/CMakeLists.txt%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/248%23discussion_r40314653%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22unrelated%20change%22%2C%20%22created_at%22%3A%20%222015-09-24T12%3A50%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-28T11%3A39%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/functional-tests/CMakeLists.txt%3AL42-49%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/cc6565%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9692938%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/cc6565'></a>

- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/248#issuecomment-142637504'>General Comment</a></b>
- <a href='https://github.com/miguelgaio'></a> @dawagner @krocard @OznOg @cc6565 please review
- <a href='https://github.com/miguelgaio'></a> :+1:<a href='#crh-update-none'></a>
- <a href='https://github.com/dawagner'></a> nice job :)
- <a href='https://github.com/dawagner'></a> @krocard or @OznOg please review. I'll merge when there is another +1.
- [x] <a href='#crh-comment-Pull d14954fb9e808b40ee07233c4f805e8f3962e1b1 xmlserializer/XmlDocSource.cpp 78'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/248#discussion_r40217333'>File: xmlserializer/XmlDocSource.cpp:L225-232</a></b>
- <a href='https://github.com/krocard'></a> all those free could be removed with unique_ptr...
- <a href='https://github.com/miguelgaio'></a> I'm not sure, xmlFree is not delete or free()
- <a href='https://github.com/krocard'></a> unique_ptr can be provided with a custom deleter (default is delete, not
free).
- [x] <a href='#crh-comment-Pull 7a207c7fd3341ebaf18501bf9f6b2f3d0efb0372 test/CMakeLists.txt 4'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/248#discussion_r40314448'>File: test/CMakeLists.txt:L26-34</a></b>
- <a href='https://github.com/dawagner'></a> It could be worth copying this directory in the legacy functional test dir too (or instead). And if so, it should be done in `functional-tests-legacy/CMakeLists.txt`.
You would avoid the `../../../../../../` in the XML files below.
EDIT: and/or use `configure_file` on all the XML files contained in the legacy functional tests.
- [x] <a href='#crh-comment-Pull 7a207c7fd3341ebaf18501bf9f6b2f3d0efb0372 test/functional-tests/CMakeLists.txt 4'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/248#discussion_r40314653'>File: test/functional-tests/CMakeLists.txt:L42-49</a></b>
- <a href='https://github.com/dawagner'></a> unrelated change
- [x] <a href='#crh-comment-Pull 7a207c7fd3341ebaf18501bf9f6b2f3d0efb0372 xmlserializer/XmlDocSource.h 5'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/248#discussion_r40315214'>File: xmlserializer/XmlDocSource.h:L57-75</a></b>
- <a href='https://github.com/dawagner'></a> This constructor seems unused. If so, you may remove it altogether (in a separate patch).
- <a href='https://github.com/miguelgaio'></a> Used by XmlMemorySource
- [x] <a href='#crh-comment-Pull 49ffd604786e225e8396aa4a2aa13d658fd40f81 parameter/ParameterMgr.cpp 93'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/248#discussion_r40546172'>File: parameter/ParameterMgr.cpp:L597-604</a></b>
- <a href='https://github.com/dawagner'></a> "URI" or "Uri" ? Please choose one and stick to it. (I'd choose "Uri" because even though it's an acronym, we have chose lowerCamelCase style. EDIT: and because we already have "mk**Xml**Doc")

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/248?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/248?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/248?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/248'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/248'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by miguelgaio** [comment]
@dawagner @krocard @OznOg @cc6565 please review

File: xmlserializer/XmlDocSource.cpp [hunk 1]
```diff
@@ -216,6 +225,7 @@ bool CXmlDocSource::isInstanceDocumentValid()
     xmlSchemaFree(pSchema);
     xmlSchemaFreeParserCtxt(pParserCtxt);
     xmlFreeDoc(pSchemaDoc);
+    xmlFree(schemaURI);
```
Comment (krocard) [inline xmlserializer/XmlDocSource.cpp]: all those free could be removed with unique_ptr...

Comment (miguelgaio) [inline xmlserializer/XmlDocSource.cpp]: I'm not sure, xmlFree is not delete or free()

Comment (krocard) [inline xmlserializer/XmlDocSource.cpp]: unique_ptr can be provided with a custom deleter (default is delete, not 
free).

On 09/23/2015 05:34 PM, miguelgaio wrote:
>
> In xmlserializer/XmlDocSource.cpp 
> <https://github.com/01org/parameter-framework/pull/248#discussion_r40218494>:
>
> > @@ -216,6 +225,7 @@ bool CXmlDocSource::isInstanceDocumentValid()
> >      xmlSchemaFree(pSchema);
> >      xmlSchemaFreeParserCtxt(pParserCtxt);
> >      xmlFreeDoc(pSchemaDoc);
> > +    xmlFree(schemaURI);
>
> I'm not sure, xmlFree is not delete or free()
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/01org/parameter-framework/pull/248/files#r40218494>.
>

---------------------------------------------------------------------
Intel Corporation SAS (French simplified joint stock company)
Registered headquarters: "Les Montalets"- 2, rue de Paris, 
92196 Meudon Cedex, France
Registration Number:  302 456 199 R.C.S. NANTERRE
Capital: 4,572,000 Euros

This e-mail and any attachments may contain confidential material for
the sole use of the intended recipient(s). Any review or distribution
by others is strictly prohibited. If you are not the intended
recipient, please contact the sender and delete all copies.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -26,6 +26,8 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+file(COPY ${CMAKE_SOURCE_DIR}/Schemas/ DESTINATION ${CMAKE_BINARY_DIR}/Schemas)
```
Comment (dawagner) [inline test/CMakeLists.txt]: It could be worth copying this directory in the legacy functional test dir too (or instead). And if so, it should be done in `functional-tests-legacy/CMakeLists.txt`.

File: test/functional-tests/CMakeLists.txt [hunk 3]
```diff
@@ -42,6 +42,7 @@ if(BUILD_TESTING)
     include_directories("${CMAKE_CURRENT_SOURCE_DIR}/include")
     include_directories("${PROJECT_SOURCE_DIR}/utility")
 
+
```
Comment (dawagner) [inline test/functional-tests/CMakeLists.txt]: unrelated change

File: xmlserializer/XmlDocSource.h [hunk 4]
```diff
@@ -57,20 +57,18 @@ class CXmlDocSource : private utility::NonCopyable
       * @param[in] pRootNode a pointer to the root element of the document.
       * @param[in] bValidateWithSchema a boolean that toggles schema validation
       */
-    CXmlDocSource(_xmlDoc* pDoc, bool bValidateWithSchema = false, _xmlNode* pRootNode = NULL);
+    CXmlDocSource(_xmlDoc* pDoc, bool bValidateWithSchema, _xmlNode* pRootNode);
```
Comment (dawagner) [inline xmlserializer/XmlDocSource.h]: This constructor seems unused. If so, you may remove it altogether (in a separate patch).

Comment (miguelgaio) [inline xmlserializer/XmlDocSource.h]: Used by XmlMemorySource

File: xmlserializer/XmlDocSource.cpp (hunk 1 above)
Comment (miguelgaio) [inline xmlserializer/XmlDocSource.cpp]: :+1:<a href='#crh-update-none'></a>

**Comment by miguelgaio** [comment]
:+1:<a href='#crh-update-none'></a>

File: test/CMakeLists.txt (hunk 2 above)
Comment (miguelgaio) [inline test/CMakeLists.txt]: :+1:<a href='#crh-update-none'></a>

File: xmlserializer/XmlDocSource.h (hunk 4 above)
Comment (miguelgaio) [inline xmlserializer/XmlDocSource.h]: :+1:<a href='#crh-update-none'></a>

File: test/functional-tests/CMakeLists.txt (hunk 3 above)
Comment (miguelgaio) [inline test/functional-tests/CMakeLists.txt]: :+1:<a href='#crh-update-complete'></a>

File: parameter/ParameterMgr.cpp [hunk 5]
```diff
@@ -615,10 +597,7 @@ bool CParameterMgr::loadSettingsFromConfigFile(string& strError)
     CConfigurableDomains* pConfigurableDomains = getConfigurableDomains();
 
     // Get Xml configuration domains file name
-    string strXmlConfigurationDomainsFilePath = pConfigurableDomainsFileLocation->getFilePath(_strXmlConfigurationFolderPath);
-
-    // Get Xml configuration domains folder
-    string strXmlConfigurationDomainsFolder = pConfigurableDomainsFileLocation->getFolderPath(_strXmlConfigurationFolderPath);
+    string configurationDomainsUri = CXmlDocSource::mkURI(_strXmlConfigurationUri, pConfigurableDomainsFileLocation->getUri());
```
Comment (dawagner) [inline parameter/ParameterMgr.cpp]: "URI" or "Uri" ? Please choose one and stick to it. (I'd choose "Uri" because even though it's an acronym, we have chose lowerCamelCase style)

**Comment by dawagner** [comment]
nice job :)

File: parameter/ParameterMgr.cpp (hunk 5 above)
Comment (miguelgaio) [inline parameter/ParameterMgr.cpp]: :+1:

**Comment by dawagner** [comment]
@krocard or @OznOg please review. I'll merge when there is another +1.

**Comment by cc6565** [comment]
:shipit:
